File: level1/완주하지 못한 선수/jihyung.java
import java.util.HashMap;
class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String,Integer> completion_participant_count = new HashMap<>();
        for(String runner :completion){
            if(completion_participant_count.get(runner) == null){
                completion_participant_count.put(runner, 1);
            }else{
                completion_participant_count.put(runner, completion_participant_count.get(runner) +1);
            }
        }
        
        String answer = "";
        for(String runner :participant){
            if(completion_participant_count.containsKey(runner) && completion_participant_count.get(runner) >= 1){
                completion_participant_count.put(runner, completion_participant_count.get(runner) -1);
            }else{
                answer = runner;
            }
        }
        
        return answer;
    }
}

File: programmers/level1/완주하지 못한 선수/jihyung.py
def solution(participant, completion):
    # 완주자의 이름과 수를 저장함
    completion_participant_count = {}
    for runner in completion:
        if runner not in completion_participant_count:
            completion_participant_count[runner] = 1
        else:
            completion_participant_count[runner] += 1

    for runner in participant:
        # 주자가 완주자목록에 들어있고 주자의 수가 1이상이면 ok
        if runner in completion_participant_count and completion_participant_count[runner] >= 1:
            completion_participant_count[runner] -= 1

        # 아닐경우 완주하지 못한 주자입니다.
        else:
            return runner


print(solution(["leo", "kiki", "eden"], ["eden", "kiki"]))  # "leo"
